### DISCUSSION TITLE
Frequency-dependent Q

### DISCUSSION THREAD

**Discussion by jf-huang** [geodynamics/sw4#188] [General]
Hi, I looked at the part of code that fits a constant Q with the least squares scheme. It seems to me the same routine can be used to fit an arbitrary frequency-dependent Q by replacing Qs (Qp) with the desired Q values on the following line of code:
https://github.com/geodynamics/sw4/blob/efbf019501728ea89313f5d94ab982173954c574/src/EW.C#LL7939C1-L7939C25
I understand the number of mechanisms required to achieve a intended Q law within a acceptable error bound may be different that the implemented and tested constant Q case. I was wondering whether this thought is true and whether there are resultant pitfalls? Thanks a lot!

**Comment by andersp** [comment]
I don't know what will happen if you do that, but I encourage you to try and report back. If it works, you can issue a pull request

**Comment by jf-huang** [comment]
Thanks @andersp , I am trying it out and it seems to be working. I will make a PR soon.
